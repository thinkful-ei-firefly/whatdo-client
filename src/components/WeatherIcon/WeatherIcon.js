import React from 'react'

export default class WeatherIcon extends React.Component {

  timestamp () {
    const date = new Date(this.props.dt)
    const [time, period] = date.toLocaleTimeString().split(' ')
    return `${time.split(':')[0]} ${period}`
  }

  render () {

    return (
      <div className="weathericon">
        <h3 className="weathericon-time">{this.timestamp()}</h3>
        <img className="weathericon-icon" src={this.props.iconUrl} alt={this.props.description}/>
        <p className="weathericon-temperature high">High:{this.props.temp_hi}</p>
        <p className="weathericon-temperature low">Low:{this.props.temp_low}</p>
      </div>
    );
  }
}