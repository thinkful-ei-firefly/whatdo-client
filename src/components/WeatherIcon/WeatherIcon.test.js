import React from 'react';
import ReactDOM from 'react-dom';
import WeatherIcon from './WeatherIcon';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WeatherIcon />, div);
  ReactDOM.unmountComponentAtNode(div);
});
