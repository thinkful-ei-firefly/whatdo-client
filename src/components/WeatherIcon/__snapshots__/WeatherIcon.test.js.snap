// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon renders properly 1`] = `
<div
  className="weathericon"
>
  <h3
    className="weathericon-time"
  >
    8 PM
  </h3>
  <img
    alt="hot"
    className="weathericon-icon"
    src=""
  />
  <p
    className="weathericon-temperature high"
  >
    High: 
    99
  </p>
  <p
    className="weathericon-temperature low"
  >
    Low: 
    0
  </p>
</div>
`;
